app/src/main/java/com/mk/tasky/authentication/registration/presentation/RegistrationViewModel.kt

```diff
@@ -0,0 +1,27 @@
+package com.mk.tasky.authentication.registration.presentation
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.mk.tasky.authentication.domain.AuthenticationRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class RegistrationViewModel @Inject constructor(
+    private val repository: AuthenticationRepository
+) : ViewModel() {
+    init {
+        viewModelScope.launch {
+            try {
+                repository.register("Testing", "testing@mk.com", "teasASDF1").onSuccess {
+                    println(it)
+                }.onFailure {
+                    println(it)
+                }
+            } catch (e: Exception) {
+                println(e)
+            }
```

Comment:
In which case would this try/catch be necessary? Doesn't the `register` function already handle this?

Reply:
CancellationException throws back the exception, so I would need to handle those cases. Right?

Reply:
CancellationExceptions are fully handled by the coroutine framework already, you don't need to worry about them (just make sure you don't accidentally catch them in your suspend functions). Your app won't crash if a CancellationException happens

Reply:
Awesome, didn't knew that.
Thanks!